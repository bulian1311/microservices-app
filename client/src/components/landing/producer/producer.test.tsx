import React from "react";
import { renderHook } from "@testing-library/react-hooks";
import { render, fireEvent } from "@testing-library/react";
import useStore from "../../../hooks/use-store.hook";
import Producer from "./producer.component";

let storeHook = renderHook(() => useStore());

const fetchProductsSpy = jest.spyOn(
  storeHook.result.current.productStore,
  "fetchProducts"
);

const setSearchQuerySpy = jest.spyOn(
  storeHook.result.current.filterStore,
  "setSearchQuery"
);

test("Producer component, onClick event.", () => {
  const documentBody = render(<Producer producer="party2go" />);
  const elem = documentBody.getByTestId("producer");

  fireEvent.click(elem);

  expect(setSearchQuerySpy).toBeCalled();
  expect(fetchProductsSpy).toBeCalled();
});

test("Producer match Snapshot 1.", () => {
  const { baseElement } = render(<Producer producer="party2go" />);
  expect(baseElement).toMatchSnapshot();
});

test("Producer match Snapshot 2.", () => {
  const { baseElement } = render(<Producer producer="arenda-attrakcionov" />);
  expect(baseElement).toMatchSnapshot();
});

test("Producer match Snapshot 3.", () => {
  const { baseElement } = render(<Producer producer="batutvarendu" />);
  expect(baseElement).toMatchSnapshot();
});

test("Producer match Snapshot 4.", () => {
  const { baseElement } = render(<Producer producer="eventomania" />);
  expect(baseElement).toMatchSnapshot();
});

test("Producer match Snapshot 5.", () => {
  const { baseElement } = render(<Producer producer="funevents" />);
  expect(baseElement).toMatchSnapshot();
});

test("Producer match Snapshot 6.", () => {
  const { baseElement } = render(<Producer producer="pro-interactive" />);
  expect(baseElement).toMatchSnapshot();
});
