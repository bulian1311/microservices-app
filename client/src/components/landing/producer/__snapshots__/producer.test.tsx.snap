// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Producer match Snapshot 1. 1`] = `
<body>
  <div>
    <div
      class="sc-bdfBwQ cSkijj"
      data-testid="producer"
    >
      <img
        class="sc-gsTCUz hBKqoK"
        src="test-file-stub"
        title="party2go"
      />
    </div>
  </div>
</body>
`;

exports[`Producer match Snapshot 2. 1`] = `
<body>
  <div>
    <div
      class="sc-bdfBwQ cSkijj"
      data-testid="producer"
    >
      <img
        class="sc-gsTCUz hBKqoK"
        src="test-file-stub"
        title="arenda-attrakcionov"
      />
    </div>
  </div>
</body>
`;

exports[`Producer match Snapshot 3. 1`] = `
<body>
  <div>
    <div
      class="sc-bdfBwQ cSkijj"
      data-testid="producer"
    >
      <img
        class="sc-gsTCUz hBKqoK"
        src="test-file-stub"
        title="batutvarendu"
      />
    </div>
  </div>
</body>
`;

exports[`Producer match Snapshot 4. 1`] = `
<body>
  <div>
    <div
      class="sc-bdfBwQ cSkijj"
      data-testid="producer"
    >
      <img
        class="sc-gsTCUz hBKqoK"
        src="test-file-stub"
        title="eventomania"
      />
    </div>
  </div>
</body>
`;

exports[`Producer match Snapshot 5. 1`] = `
<body>
  <div>
    <div
      class="sc-bdfBwQ cSkijj"
      data-testid="producer"
    >
      <img
        class="sc-gsTCUz hBKqoK"
        src="test-file-stub"
        title="funevents"
      />
    </div>
  </div>
</body>
`;

exports[`Producer match Snapshot 6. 1`] = `
<body>
  <div>
    <div
      class="sc-bdfBwQ cSkijj"
      data-testid="producer"
    >
      <img
        class="sc-gsTCUz hBKqoK"
        src="test-file-stub"
        title="pro-interactive"
      />
    </div>
  </div>
</body>
`;
